frontend/src/components/pages/logHours.tsx

```diff
@@ -72,6 +80,63 @@ export default function LogHours({ eventData }: LogHoursProps) {
     return event._id === eventId;
   });
 
+  const addToUser = async (id: string) => {
+    await fetch(`http://localhost:3001/users/${currentUser._id}`, {
+      method: 'PUT',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({ shiftId: id }),
+    });
+  };
+
+  const updateUserHours = async () => {
+    await fetch(`http://localhost:3001/users/${currentUser._id}/test`, {
+      method: 'PUT',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({ numHours: hours }),
+    });
+  };
+
+  const addToEvent = async (id: string) => {
+    await fetch(`http://localhost:3001/events/${eventId}`, {
+      method: 'PUT',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({ shiftId: id }),
+    });
+  };
+
+  const addShift = async () => {
+    const shift = {
+      event: eventId,
+      hours,
+      user: currentUser._id,
+      userName: currentUser.name,
+    };
+
+    console.log(shift);
+
+    await fetch(`http://localhost:3001/shifts`, {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify(shift),
+    })
+      .then((res) => res.json())
+      .then((data) => {
+        const id = data._id;
+        addToUser(id);
+        addToEvent(id);
+        setPastShifts((prev: Shift[]) => [...prev, data]);
+        // TODO: if admin, update allShifts for the volunteer log
+        setAllShifts((prev: Shift[]) => [...prev, data]);
+      })
```

Comment:
Instead of your original `setPastShifts([...pastShifts, data])`, I changed to the preferred method of `setPastShifts((prev) => [...prev, data])`. The callback automatically provides the previous state (named prev in this case) so we don't have to pass `pastShifts` as props to this component.